dataflow/gemma-flex-template/README.md

```diff
@@ -0,0 +1,179 @@
+# RunInference on Dataflow streaming with Gemma and Flex Templates
+
+Gemma is a family of lightweight, state-of-the-art open models built from research and technology used to create the Gemini models.
+You can use Gemma models in your Apache Beam inference pipelines with the `RunInference` transform.
+
+This example demonstrates how to use a Gemma model running on Pytorch in a streaming Dataflow pipeline that has Pub/Sub sources and sinks. This pipeline
+is deployed by using Flex Templates.
+
+For more information about using RunInference, see [Get started with AI/ML pipelines](https://beam.apache.org/documentation/ml/overview/) in the Apache Beam documentation.
+
+## Before you begin
+
+Make sure you have followed the [Dataflow setup instructions](/dataflow/README.md).
+
+Follow the steps in this section to create the necessary environment to run this workflow.
+
+### Enable Google Cloud services
+
+This workflow uses multiple Google Cloud products, including Dataflow, Pub/Sub, Google Cloud Storage, and Artifact Registry. Before you start the workflow, create a Google Cloud project that has the following services enabled:
+
+* Dataflow
+* Pub/Sub
+* Compute Engine
+* Cloud Logging
+* Google Cloud Storage
+* Google Cloud Storage JSON
+* Cloud Build
+* Datastore
+* Cloud Resource Manager
+* Artifact Registry
+
+Using these services incurs billing charges.
+
+Your Google Cloud project also needs to have Nvidia L4 GPU quota. For more information, see [GPU quota](https://cloud.google.com/compute/resource-usage#gpu_quota) in the Google Cloud documentation.
+
+### Download and save the model
+
+Save a version of the Gemma 2B model. Downloaded the model from [Kaggle](https://www.kaggle.com/models/google/gemma/pyTorch/1.1-2b-it). This download is a `.tar.gz` archive. Extract the archive into a directory and name it `pytorch_model`. 
+
+### Create a cloud storage bucket
+
+Click [here to create a GCS bucket](https://console.cloud.google.com/storage/create-bucket) or run the following commands:
+
+```sh
+export GCS_BUCKET="your--bucket"
+gsutil mb gs://$GCS_BUCKET
+```
+
+Make sure your GCS bucket name does __not__ include the `gs://` prefix
+
+### Create a custom container
+
+To build a custom container, use Docker. This repository contains a Dockerfile that you can use to build your custom container. To build and push a container to Artifact Registry by using Docker or Cloud Build, follow the instructions in the [Build and push the image](https://cloud.google.com/dataflow/docs/guides/build-container-image#build_and_push_the_image) section of "Build custom container images for Dataflow" in the Google Cloud documentation.
```

Comment:
question: just curious if any supported OCI image would do. Artifact Registry supports them. Then, this is more of a We've provided a sample Dockerfile to showcase building a custom container. There are other tools folks use for building containers and it's better to not imply the specific tool choice if it's not necessary. I might recommend dropping the first sentence of "To build a custom container, use Docker"